Content/GUI/Elements/UIHoverTooltipImage.cs

```diff
@@ -0,0 +1,77 @@
+using ReLogic.Content;
+using Terraria.GameContent.UI.Elements;
+using Terraria.Localization;
+
+namespace PathOfTerraria.Content.GUI.Elements;
+
+/// <summary>
+///     Provides a <see cref="UIImage" /> that displays a tooltip on hover
+///     and contains hover transform effects.
+/// </summary>
+public class UIHoverTooltipImage : UIImage
+{
+	/// <summary>
+	///     The target rotation for this image when it's being hovered by the mouse.
+	/// </summary>
+	/// <remarks>
+	///     Defaults to <c>0f</c>.
+	/// </remarks>
+	public float ActiveRotation = 0f;
+
+	/// <summary>
+	///     The target scale for this image when it's being hovered by the mouse.
+	/// </summary>
+	/// <remarks>
+	///     Defaults to <c>1f</c>.
+	/// </remarks>
+	public float ActiveScale = 1f;
+
+	/// <summary>
+	///     The target rotation for this image when it's not being hovered by the mouse.
+	/// </summary>
+	/// <remarks>
+	///     Defaults to <c>0f</c>.
+	/// </remarks>
+	public float InactiveRotation = 0f;
+
+	/// <summary>
+	///     The target scale for this image when it's not being hovered by the mouse.
+	/// </summary>
+	/// <remarks>
+	///     Defaults to <c>1f</c>.
+	/// </remarks>
+	public float InactiveScale = 1f;
+
+	/// <summary>
+	///     The localization key for the image's tooltip.
+	/// </summary>
+	public string Key;
+
+	/// <summary>
+	///     The smoothness used to perform scale/rotation interpolations.
+	/// </summary>
+	/// <remarks>
+	///     Defaults to <c>0.3f</c>. Ranges from <c>0f</c> - <c>1f</c>.
+	/// </remarks>
+	public float Smoothness = 0.3f;
+
+	public UIHoverTooltipImage(Asset<Texture2D> texture, string key) : base(texture)
+	{
+		Key = key;
+	}
+
+	public override void Update(GameTime gameTime)
+	{
+		base.Update(gameTime);
+
+		Rotation = MathHelper.SmoothStep(Rotation, IsMouseHovering ? ActiveRotation : InactiveRotation, Smoothness);
+		ImageScale = MathHelper.SmoothStep(ImageScale, IsMouseHovering ? ActiveScale : InactiveScale, Smoothness);
+
+		if (!IsMouseHovering)
+		{
+			return;
+		}
+
+		Main.instance.MouseText(Language.GetTextValue(Key));
```

Comment:
Not an issue, but we have a `Tooltip` class in PoT that shows a name and a tooltip + backing panel. 100% preference and doesn't change any functionality, but that may be preferred in some cases.

Reply:
> Not an issue, but we have a `Tooltip` class in PoT that shows a name and a tooltip + backing panel. 100% preference and doesn't change any functionality, but that may be preferred in some cases.

Oh huh, this would've been nice. In the future I hope we can get to do some UI refactoring to keep everything centralized and uniform, so issues like this dont occur more often.

Reply:
It'd also help to have more documentation on in-house stuff. There's so much in the codebase already, such as that Tooltip class, but I wouldn't know of it without just accidentally passing by it while looking for something else.